src/com/sap/piper/CfUtils.groovy

```diff
@@ -0,0 +1,40 @@
+package com.sap.piper
+
+def getXsuaaCredentials(String apiEndpoint, String org, String space, String credentialsId, String appName, boolean verbose){
```

Comment:
I would suggest to break this down into small reusable methods like
- getAuthEndpoint
- getToken
- getOrgList
- getAppList
- getEnvironment

Reply:
I agree. The code would become much clearer. Now you have a variable named `org_guid`, but in fact it contains the response JSON. Then you extract that into `responseJson`, which is in fact the unmarshalled Map and in `responseJson.resources[0].guid` you have the "actual" guid which is somewhat hidden in the request parameters. Refactoring this into separate methods, you would return `responseJson.resources[0].guid` from `getOrgGUID()`, and use this value for calling the next method.